tests/test_views.py

```diff
@@ -608,3 +613,115 @@ async def test_invoice_ws(async_client, token: str):
         await websocket.connect()
         await check_ws_response(websocket)
+
+
+def test_create_invoice_discount(client: TestClient, token: str):
+    # create discount
+    new_discount = {"name": "apple", "percent": 50, "end_date": "2099-12-31 00:00:00.000000"}
+    create_discount_resp = client.post("/discounts", json=new_discount, headers={"Authorization": f"Bearer {token}"})
+    assert create_discount_resp.status_code == 200
+    discount_id = create_discount_resp.json()["id"]
+    # create product
+    new_product = {"name": "apple", "price": 0.80, "quantity": 1.0, "store_id": 2, "discounts": [discount_id]}
+    create_product_resp = client.post(
+        "/products", data={"data": json_module.dumps(new_product)}, headers={"Authorization": f"Bearer {token}"}
+    )
+    assert create_product_resp.status_code == 200
+    product_id = create_product_resp.json()["id"]
+    invoice_resp = client.post(
+        "/invoices",
+        json={"store_id": 2, "price": 0.5, "products": [product_id], "discount": discount_id},
+        headers={"Authorization": f"Bearer {token}"},
+    )
+    assert invoice_resp.status_code == 200
+    assert {
+        "price": 0.5,
+        "store_id": 2,
+        "discount": discount_id,
+        "products": [product_id],
+    }.items() < invoice_resp.json().items()
+    assert (
+        client.delete(f"/invoices/{invoice_resp.json()['id']}", headers={"Authorization": f"Bearer {token}"}).status_code
+        == 200
+    )
+
+
+@pytest.mark.parametrize("order_id,expect_status_code", [(-1, 404), (10, 200)])
+def test_get_invoice_by_order_id(client: TestClient, token: str, order_id: int, expect_status_code):
+    resp = None
+    if expect_status_code == 200:
+        resp = client.post(
+            "/invoices",
+            json={"store_id": 2, "price": 0.98, "order_id": str(order_id)},
+            headers={"Authorization": f"Bearer {token}"},
+        )
+        assert resp.status_code == 200
+    assert (
+        client.get(f"/invoices/order_id/{order_id}", headers={"Authorization": f"Bearer {token}"}).status_code
+        == expect_status_code
+    )
+    if resp:
+        assert client.delete(f"/invoices/{resp.json()['id']}", headers={"Authorization": f"Bearer {token}"}).status_code == 200
+
+
+def test_get_max_product_price(client: TestClient, token: str):
+    # create product
+    price = 999999.0
+    new_product = {"name": "apple", "price": price, "quantity": 1.0, "store_id": 2}
+    create_product_resp = client.post(
+        "/products", data={"data": json_module.dumps(new_product)}, headers={"Authorization": f"Bearer {token}"}
+    )
+    assert create_product_resp.status_code == 200
+    maxprice_resp = client.get(f"/products/maxprice?store={2}", headers={"Authorization": f"Bearer {token}"})
+    assert maxprice_resp.status_code == 200
+    assert maxprice_resp.json() == price
+
+
+def test_create_product_with_image(client: TestClient, token: str):
+    image = (
```

Comment:
Might be good to instead load that as fixture from file (i.e. `tests/fixtures/image.png`) instead of placing in test file directly